tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/CalDavEventAttendanceRepository.java

```diff
@@ -0,0 +1,182 @@
+                                                                         BlobIds calendarEventBlobIds) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return Flux.fromIterable(JavaConverters.seqAsJavaList(calendarEventBlobIds.value()))
+            .flatMap(blobId -> getEventUid(blobId.value(), systemMailboxSession)
+                .flatMap(eventUid ->
+                    toDavUser(username)
+                        .flatMap(davUser -> davClient.getCalendarObjectContainingVEvent(davUser, eventUid)))
+                .map(calendarObject -> getAttendanceStatusFromCalendarObject(
+                    BlobId.of(blobId.value()).get(), calendarObject, username, systemMailboxSession)))
+            .reduce(CalendarEventAttendanceResults$.MODULE$.empty(), CalendarEventAttendanceResults$.MODULE$::merge);
+    }
+
+    private CalendarEventAttendanceResults getAttendanceStatusFromCalendarObject(BlobId blobId,
+        DavCalendarObject calendarObject, Username username, MailboxSession mailboxSession) {
+
+        try {
+            List<CalendarEventParsed> events = JavaConverters.asJava(CalendarEventParsed.from(calendarObject.calendarData()));
+
+            if (events.size() != 1) {
+                if (events.isEmpty()) {
+                    LOGGER.debug("""
+                No VEvents found in calendar object.
+                Returning empty attendance results.
+                """);
+                    return CalendarEventAttendanceResults.empty();
+                } else {
+                    LOGGER.debug("""
+                Expected exactly one VEvent, but found {} entries. Using the first VEvent.
+                This may indicate unhandled recurrent events or a malformed calendar object.
+                VEvents: {}
+                """, events.size(), events);
+                }
+            }
+
+            CalendarEventParsed targetVEvent = events.getFirst();
+
+            return targetVEvent.participants()
+                .findParticipantByMailTo(username.asMailAddress().toString())
+                .flatMap(CalendarAttendeeField::participationStatus)
+                .map(p -> CalendarEventAttendanceResults$.MODULE$.done(
+                    new EventAttendanceStatusEntry(
+                        blobId.value().value(),
+                        AttendanceStatus.fromCalendarAttendeeParticipationStatus(p).orElseThrow())
+                )).getOrElse(() -> CalendarEventAttendanceResults$.MODULE$.notFound(blobId));
+
+        } catch (Exception e) {
+            return CalendarEventAttendanceResults$.MODULE$.notDone(blobId, e, mailboxSession);
+        }
+    }
+
+    private Mono<DavUser> toDavUser(Username username) {
+        try {
+            return davUserProvider.provide(username.asMailAddress());
+        } catch (AddressException e) {
+            LOGGER.debug("Failed to get user DAV information", e);
+            return Mono.empty();
+        }
+    }
```

Comment:
Error management should be done within davUserProvider.provide.

Having reactive methods throwing is an anti pattern preventing us from inlining the call.